crypto/asn1/tasn_dec.c

```diff
@@ -706,7 +706,7 @@ static int asn1_d2i_ex_primitive(ASN1_VALUE **pval,
     const unsigned char *p;
     BUF_MEM buf = { 0, NULL, 0, 0 };
     const unsigned char *cont = NULL;
-    long len;
+    size_t len;
```

Comment:
I assume you changed it to size_t because of the BUF_MEM usage. But long is used for lengths all around in this call and thus the length will fit long, so I rather propose to explicitly cast to long or size_t where needed.